granian/server.py

```diff
@@ -577,11 +588,28 @@ def _serve_with_reloader(self, spawn_target, target_loader):
             logger.error('Using --reload requires the granian[reload] extra')
             sys.exit(1)
 
-        reload_path = Path.cwd()
+        # Use given or default filter rules
+        reload_filter = self.reload_filter or watchfiles.filters.DefaultFilter
+        # Extend `reload_filter` with explicit args
+        ignore_dirs = (*reload_filter.ignore_dirs, *self.reload_ignore_dirs)
+        ignore_entity_patterns = (
+            *reload_filter.ignore_entity_patterns,
+            *self.reload_ignore_entity_patterns,
+        )
+        ignore_paths = (*reload_filter.ignore_paths, *self.reload_ignore_paths)
+        # Construct new filter
+        reload_filter = watchfiles.filters.DefaultFilter(
+            ignore_dirs=ignore_dirs, ignore_entity_patterns=ignore_entity_patterns, ignore_paths=ignore_paths
+        )
```

Comment:
I get from this code that there's no way to actually *override* `watchfiles` defaults from the CLI, right?

Given my suggestion on using `None` as default for all these variables, maybe we can use the `DefaultFilter` values only if `self` items are not `None`, wdyt?

Reply:
We can't (comfortably) override an existing `*Filter` obj because of the preprocessing that happens in the `__init__`.
We could override the public members, call the `__init__` again or create a new obj like I did.
Calling the `__init__` again might be prettier, to be honest.

We should probably first think about what the expected behaviour from CLI is. 
1. Extending defaults
2. Replacing defaults

I thought extending the `DefaultFilter` makes sense because normally, one wouldn't want to miss out on the defaults they provide.
https://github.com/samuelcolvin/watchfiles/blob/main/watchfiles/filters.py#L75-L99

If someone **really** wants to replace the `DefaultFilter` defaults, they can use Granian from Python and provide an obj to `reload_filter`.

Reply:
> If someone really wants to replace the DefaultFilter defaults, they can use Granian from Python and provide an obj to reload_filter.